### DISCUSSION TITLE
Bootstrap recommended best practices?

### DISCUSSION THREAD

**Discussion by Mic1** [twbs/bootstrap#36451] [Q&A]
In currrent docs, including latest 5.2, under Customize - Sass, there is

"Get started with Bootstrap via npm with our starter project! Head to the [twbs/bootstrap-npm-starter](https://github.com/twbs/bootstrap-npm-starter) template repository to see how to build and customize Bootstrap in your own npm project. Includes Sass compiler, Autoprefixer, Stylelint, PurgeCSS, and Bootstrap Icons."

The example project is in 4 with note to upgrade to 5, but it is referenced by 5.2 docs, and what it shows is a custom component "_icon-list.scss" imported by starter.scss. The identity of this custom component is completely encapsulated in _icon-list.scss, as the html is vanilla with class="icon-list" being applied to the ul tag.

Because _icon-list.scss is imported in starter.scss, which imports bootstrap above it - functions, variables, mixins etc, _icon-list has access to all the bootstrap variables. This example seems to suggest creating all custom components as scss in the scss (styles) folder where they can get to everything and writing pure html. which appears to have some elegance - readable html, inheritance:

`.icon-list2 {
  @extend .icon-list;

  background-color: aqua;
}`,

all components in same place etc ...... but it also seems to ignore 99% of what the bootstrap docs consist of - <div class="container-fluid"> and the other nine yards.

In summary it is a completely different way of doing things than the rest of the bootstrap docs describe. Am interested in comments.
